src/components/ReportActionItem/MoneyRequestPreview.tsx

```diff
@@ -229,18 +219,18 @@ function MoneyRequestPreview(props) {
             return translate('iou.receiptScanning');
         }
 
-        if (TransactionUtils.hasMissingSmartscanFields(props.transaction)) {
+        if (TransactionUtils.hasMissingSmartscanFields(transaction)) {
             return Localize.translateLocal('iou.receiptMissingDetails');
         }
 
         return CurrencyUtils.convertToDisplayString(requestAmount, requestCurrency);
     };
 
-    const getDisplayDeleteAmountText = () => {
-        const {amount, currency} = ReportUtils.getTransactionDetails(props.action.originalMessage);
+    const getDisplayDeleteAmountText = (): string => {
+        const {amount, currency} = ReportUtils.getTransactionDetails(action.originalMessage as OnyxTypes.Transaction) ?? {};
```

Comment:
I have some doubts regarding this assertion
1. `action.originalMessage` looks like this
![image](https://github.com/Expensify/App/assets/23176449/23678e6f-5d00-464f-8a7b-7344a80725e1)
2. Updating the type of `ReportUtils.getTransactionDetails` to accept `IOUMessage` type as well will trigger a lot of updates

Reply:
This is very very strange indeed, `IOUMessage` and `Transaction` looks like completely different things. Maybe we could create another function similar to what `getTransactionDetails` do, but to only return `amount` and `currency`?

Reply:
I've tested it more, and I guess the function can be updated to
```
    const getDisplayDeleteAmountText = (): string => {
        const iouOriginalMessage: IOUMessage | EmptyObject = action?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU ? action.originalMessage : {};
        const {amount = 0, currency = CONST.CURRENCY.USD} = iouOriginalMessage;

        return CurrencyUtils.convertToDisplayString(amount, currency);
    };
```
What do you think?